packages/components/src/utils/hooks/test/use-focus-outside.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * External dependencies
+ */
+import TestUtils from 'react-dom/test-utils';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useFocusOutside } from '../';
+import ReactDOM from 'react-dom';
+
+let wrapper, onFocusOutside;
+
+describe( 'useFocusOutside', () => {
+	let origHasFocus;
+
+	const FocusOutsideComponent = ( { onFocusOutside: callback } ) => (
+		<div { ...useFocusOutside( callback ) }>
+			<input />
+			<input type="button" />
+		</div>
+	);
+
+	// this is needed because TestUtils does not accept a stateless component.
+	// anything run through a HOC ends up as a stateless component.
+	const getTestComponent = ( WrappedComponent, props ) => {
+		class TestComponent extends Component {
+			render() {
+				return <WrappedComponent { ...props } />;
+			}
+		}
+		return <TestComponent />;
+	};
+
+	const simulateEvent = ( event, index = 0 ) => {
+		const element = TestUtils.scryRenderedDOMComponentsWithTag(
```

Comment:
I wonder can it be done with RTL?

We should ship our own version of `test-utils` otherwise, importing `react-dom` in gutenberg feels 
slightly off to me 😂.

Reply:
Looks like there are a few files that do this, I think it could be treated as a separate task to update them all.